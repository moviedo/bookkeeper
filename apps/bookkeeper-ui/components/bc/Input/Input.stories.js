export default {
  title: "base components / bc-input",
};

const Template = (_args, { argTypes }) => ({
  props: Object.keys(argTypes),
  template: '<bc-input v-bind="$props" placeholder="default data" />',
});

export const Default = Template.bind({});

export const Tablet = Template.bind({});
Tablet.parameters = {
  viewport: {
    defaultViewport: "tablet",
  },
};

export const Mobile = Template.bind({});
Mobile.parameters = {
  viewport: {
    defaultViewport: "mobile1",
  },
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const Hover = Template.bind({});
Hover.parameters = {
  pseudo: {
    hover: true,
  },
};

export const Focus = Template.bind({});
Focus.parameters = {
  pseudo: {
    focus: true,
  },
};

export const ErrorDefault = Template.bind({});
ErrorDefault.args = {
  errored: true,
};

export const ErrorFocus = Template.bind({});
ErrorFocus.parameters = { ...Focus.parameters };
ErrorFocus.args = { ...ErrorDefault.args };
