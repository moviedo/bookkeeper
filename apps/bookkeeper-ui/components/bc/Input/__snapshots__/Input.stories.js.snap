// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base components / bc-input Default smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Default smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Default smoke-test: base-components-bc-input--default 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Disabled smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       disabled="disabled"
>
`;

exports[`base components / bc-input Disabled smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       disabled="disabled"
>
`;

exports[`base components / bc-input Disabled smoke-test: base-components-bc-input--disabled 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       disabled="disabled"
>
`;

exports[`base components / bc-input ErrorDefault smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input ErrorDefault smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input ErrorDefault smoke-test: base-components-bc-input--error-default 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input ErrorFocus smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input ErrorFocus smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input ErrorFocus smoke-test: base-components-bc-input--error-focus 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
       errored="true"
>
`;

exports[`base components / bc-input Focus smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Focus smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Focus smoke-test: base-components-bc-input--focus 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Hover smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Hover smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Hover smoke-test: base-components-bc-input--hover 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Mobile smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Mobile smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Mobile smoke-test: base-components-bc-input--mobile 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Tablet smoke-test 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Tablet smoke-test 2`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;

exports[`base components / bc-input Tablet smoke-test: base-components-bc-input--tablet 1`] = `
<input data-v-1a110ac2
       type="text"
       class="default hover focus"
       placeholder="default data"
>
`;
